text/XXXX-config_revision.md

```diff
@@ -0,0 +1,268 @@
+
+In addition, we should support the convention of using `0` as a port number to indicate "any open port", allowing the operating system to choose.
+
+
+# Reference-level explanation
+[reference-level-explanation]: #reference-level-explanation
+
+### Remove stale configuration fields
+
+The following fields should be **removed**:
+
+```yaml
+API:
+  proxyforcehttps: false
+  remoteacceptsizemax: 0
+  remoteaccepttimeoutms: 0
+  remotemode: false
+  tls: false
+  urlroot: ""
+P2P:
+  autoNAT: false
+  profilereplication: full
+  bootstrapaddrs: []
+  httpgatewayaddr: ""
+  pubkey: ""
+Repo:
+  middleware: []
+Webapp:
+  analyticstoken: ""
+  enabled: true
+  entrypointhash: /ipfs/QmXofmXcQKnYTMjsfhgTGqzzLLPyS9enaHsfaRR5AxTGBK
+  entrypointupdateaddress: /ipns/webapp.qri.io
+  port: 2505
+Render:
+  defaultTemplateHash: /ipfs/QmeqeRTf2Cvkqdx4xUdWi1nJB2TgCyxmemsL3H4f1eTBaw
+  templateUpdateAddress: /ipns/defaulttmpl.qri.io
+Stats: null
+Update:
+  address: ""
+  daemonize: false
+  type: fs
+```
+
+```yaml
+API:
+  allowedorigins:
+  - electron://local.qri.io
+  - http://localhost:2505
+  enabled: true
+  address: "/ip4/0.0.0.0/tcp/2503"
+  readonly: false
+  serveremotetraffic: false
+CLI:
+  colorizeoutput: true
+  pager: ""
+Logging:
+  levels:
+    lib: info
+P2P:
+  address: "/ip4/0.0.0.0/tcp/2501"
+  addrs: null
+  enabled: true
+  peerid: QmdJQpcmGMkNtKYv5bviZ7z9kPh7uLKSMzqdSbemYs3XpH
+  privkey: <base64-encoded private key>
+  qribootstrapaddrs:
+  - /ip4/35.193.162.149/tcp/4001/ipfs/QmTZxETL4YCCzB1yFx4GT1te68henVHD1XPQMkHZ1N22mm
+  - /ip4/35.239.80.82/tcp/4001/ipfs/QmdpGkbqDYRPCcwLYnEm8oYGz2G9aUZn9WwPjqvqw3XUAc
+  - /ip4/35.225.152.38/tcp/4001/ipfs/QmTRqTLbKndFC2rp6VzpyApxHCLrFV35setF1DQZaRWPVf
+  - /ip4/35.202.155.225/tcp/4001/ipfs/QmegNYmwHUQFc3v3eemsYUVf3WiSg4RcMrh3hovA5LncJ2
+  - /ip4/35.238.10.180/tcp/4001/ipfs/QmessbA6uGLJ7HTwbUJ2niE49WbdPfzi27tdYXdAaGRB4G
+  - /ip4/35.238.105.35/tcp/4001/ipfs/Qmc353gHY5Wx5iHKHPYj3QDqHP4hVA1MpoSsT6hwSyVx3r
+  - /ip4/35.239.138.186/tcp/4001/ipfs/QmT9YHJF2YkysLqWhhiVTL5526VFtavic3bVueF9rCsjVi
+  - /ip4/35.226.44.58/tcp/4001/ipfs/QmQS2ryqZrjJtPKDy9VTkdPwdUSpTi1TdpGUaqAVwfxcNh
+Profile:
+  color: ""
+  created: "2020-02-12T04:29:18Z"
+  description: ""
+  email: brendan+test@qri.io
+  homeurl: ""
+  id: QmSyDX5LYTiwQi861F5NAwdHrrnd1iRGsoEvCyzQMUyZ4W
+  name: ""
+  peername: b5
+  photo: https://qri-user-images.storage.googleapis.com/1570029824064.png
+  poster: /ipfs/QmSN2yHp4qFy1xLCRVQNJKAjgbwHYPAho9fT7C3vAM426c
+  privkey: <base64-encoded private key>
+  thumb: ""
+  twitter: ""
+  type: peer
+  updated: "2018-04-19T18:10:46.627471799-04:00"
+RPC:
+  enabled: true
+  address: "/ip4/0.0.0.0/tcp/2504"
+Registry:
+  location: https://registry.qri.cloud
+Remote:
+  acceptsizemax: 0
+  accepttimeoutms: 0
+  allowremoves: false
+  enabled: false
+  requireallblocks: false
+Remotes: null
+Repo:
+  type: fs
+Revision: 2
+Filesystems:
+- type: ipfs
+  options:
```

Comment:
This crosses over w/ the ipfs repo move question, but are we going to give a space for folks to config ipfs ports if they so choose (if they do not want to default to 0/random)? If so, should that live in this section? Or if someone wants to do that, should we assume they will know to go into the ipfs config to change? Wondering as we used to have to manual change this when we used multiple repos on the same machine for dev purposes, and defaulting to any available port would fix this conflict anyway, but wanted to bring up if it should be surfaced/able to be controlled via config.

Reply:
>  Or if someone wants to do that, should we assume they will know to go into the ipfs config to change?

We should assume they're going to manipulate the underlying IPFS `config` file in the repo itself. I don't want to "cross wires" between the IPFS config and qri config. All fields should be set in one or the other. The `options` fields you see here _can't_ be controlled via IPFS configuration.